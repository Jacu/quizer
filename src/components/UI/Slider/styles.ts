import styled from 'styled-components';
import { PRIMARY_THEME } from '~/utils/colors';

interface ISlider {
    min: number,
    max: number,
}

export const Slider = styled.input.attrs<ISlider>(props => ({
    type: "range",
    min: props.min,
    max: props.max,
    step: 5,
}))`
    -webkit-appearance: none;
    width: 100%;
    height: 8px;
    background: lightgrey;
    outline: none;
    border-radius: 16px;
    cursor: pointer;

    ::-webkit-slider-thumb {
        -webkit-appearance: none;
        appearance: none;
        width: 20px;
        height: 20px;
        background: ${PRIMARY_THEME.BASE};
        border-radius: 50%;
    }
`;