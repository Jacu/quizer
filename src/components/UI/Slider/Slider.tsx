import React from 'react';
import * as styled from './styles';

interface ISlider {
    value: string,
    min: number,
    max: number,
    onChange: (value: string) => void,
}

const Slider: React.FC<ISlider> = (props) => {
    const handleSliderChange = (e: React.ChangeEvent<HTMLInputElement>) => {
        props.onChange(e.target.value);
    }
    return (
        <styled.Slider min={props.min} max={props.max} value={props.value} onChange={handleSliderChange}/>
    );
}

export default Slider;