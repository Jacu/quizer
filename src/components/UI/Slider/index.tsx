import React from 'react';
import * as styled from './styles';

interface ISlider {
    value: number,
    min: number,
    max: number,
    onChange: (value: number) => void,
}

const Slider: React.FC<ISlider> = ({value, min, max, onChange}) => {
    const handleSliderChange = (e: React.ChangeEvent<HTMLInputElement>) => {
        onChange(+e.target.value);
    }
    return (
        <styled.Slider min={min} max={max} value={value} onChange={handleSliderChange}/>
    );
}

export default Slider;