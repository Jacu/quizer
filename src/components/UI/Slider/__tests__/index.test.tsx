import React from 'react';
import Slider from '../';
import { render, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';

const onChange = jest.fn();
const props = {
    min: 0,
    value: 1,
    max: 2,
    onChange,
};

describe('Slider', () => {
    beforeEach(() => {
        jest.clearAllMocks();
        jest.resetAllMocks();
    });

    it('should match snapshot', () => {
        const { asFragment } = render(<Slider {...props} />);
        expect(asFragment()).toMatchSnapshot();
    });

    it('should render input', () => {
        const { container } = render(<Slider {...props} />);
        expect(container.querySelector('input')).toBeTruthy();
    });

    it('should call onChange', () => {
        const value = '5';
        const { container } = render(<Slider {...props} />);
        fireEvent.change(container.querySelector('input')!, { target: { value } });
        expect(onChange).toHaveBeenCalledWith(+value);
    });

    it('should set max range', () => {
        const max = 10;
        const { container } = render(<Slider {...props} max={max} />);
        expect(container.querySelector('input')).toHaveAttribute('max', max.toString());
    });

    it('should set min range', () => {
        const min = 1;
        const { container } = render(<Slider {...props} min={min} />);
        expect(container.querySelector('input')).toHaveAttribute('min', min.toString());
    });

    it('should set value', () => {
        const value = 1;
        const { container } = render(<Slider {...props} value={value} />);
        expect(container.querySelector('input')!.value).toBe(value.toString());
    });
});