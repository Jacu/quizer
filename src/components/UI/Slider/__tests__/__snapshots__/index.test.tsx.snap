// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slider should match snapshot 1`] = `
<DocumentFragment>
  <input
    class="sc-AxjAm juIUyd"
    max="2"
    min="0"
    step="5"
    type="range"
    value="1"
  />
</DocumentFragment>
`;
